ext/POW.c

```diff
@@ -10198,7 +10217,9 @@ init_POW(void)
    * if you tinker with the build script and start seeing nasty
    * memory-related issues, this might be the cause.
    */
+#ifdef notyet
   CRYPTO_set_mem_functions(PyMem_Malloc, PyMem_Realloc, PyMem_Free);
```

Comment:
From https://man.openbsd.org/CRYPTO_set_mem_functions#DESCRIPTION

```
DESCRIPTION
     Do not use any of the interfaces documented here.  They are provided
     purely for compatibility with legacy application code.

     CRYPTO_get_mem_functions() assigns pointers to the C library functions
     malloc(3), realloc(3), and free(3) to those of its arguments that are not
     NULL.

     CRYPTO_set_mem_functions(), CRYPTO_mem_ctrl(), CRYPTO_mem_leaks(),
     CRYPTO_mem_leaks_fp(), and CRYPTO_mem_leaks_cb() have no effect.
```

Reply:
Good catch!  I forgot about this.  I also found that documentation you cite.

Note that, unlike LibreSSL, OpenSSL 1.1 still has a working `CRYPTO_set_mem_functions()`.

For completeness, here is the Python perspective on this, from https://docs.python.org/3/c-api/memory.html:

> In most situations, however, it is recommended to allocate memory from the Python heap specifically because the latter is under control of the Python memory manager. For example, this is required when the interpreter is extended with new object types written in C. Another reason for using the Python heap is the desire to inform the Python memory manager about the memory needs of the extension module. Even when the requested memory is used exclusively for internal, highly-specific purposes, delegating all memory requests to the Python memory manager causes the interpreter to have a more accurate image of its memory footprint as a whole. Consequently, under certain circumstances, the Python memory manager may or may not trigger appropriate actions, like garbage collection, memory compaction or other preventive procedures. Note that by using the C library allocator as shown in the previous example, the allocated memory for the I/O buffer escapes completely the Python memory manager.

This is clearly "encouraging" language, as per the comment.

I am still torn about the best way forward.  On the one hand, I feel better about OpenSSL managing its memory without potential interference from Python.  On the other hand, I can easily imagine Python getting very confused about memory that isn't under its control.  The code does work fine as-is, without using the Python memory allocator.  Though I've not done any specific testing to try to catch edge cases.

For what it's worth, everyone currently running rpki.net is using the Python memory allocator...   Given that this is very much code in maintenance mode (if not outright suspended animation), I'm leaning towards providing wrappers to OpenSSL.